### DISCUSSION TITLE
Can't plot the subcircuit node voltage curve when  using  Nutmeg script

### DISCUSSION THREAD

**Discussion by LujiuZhang** [ra3xdh/qucs_s#773] [Q&A] [answered]
Dear developer,

When I use the command "write spice4qucs.tr1.plot v(X1.mz)" in Nutmeg Script to visualize the voltage change on a node in the subcircuit, I will encounter an error that is "Warning from checkvalid: vector x1.mz is not available or has zero length". I guess it is because of the destroy and reset commands, as the follow picture. Do you have any solutions?
![image](https://github.com/ra3xdh/qucs_s/assets/131197446/7a5ce166-b718-4150-998c-2ddf01d034bf)

![image](https://github.com/ra3xdh/qucs_s/assets/131197446/0e088798-c1ab-4ee2-9b1a-3a5ec47ea7fd)

**Comment by ra3xdh** [comment, marked as answer]
You are using Nutmeg script wrong. It is a separate simulation type and it knows nothing about other simulations on the schematic. You should use Nutmeg equation instead. It is different component and located in the *equations* group. Assign the subcircuit node voltage to some variable and you should be able to export in the dataset.

**Comment by ra3xdh** [comment]
Moved to Discussion/Q&A.
